Update/build lane

loremattei opened the pull request:
This PR adds GitHub releases creation to the lanes that build the beta and the final artefacts. 
It also tweaks the `code_freeze` lane to update the `release_notes.txt` file with the draft release notes so that they are available when building the first beta.  

## To Test
_1. Code freeze lane_
You can simulate a complete code freeze, but since the work the new action does is unrelated to the other steps, it can be safely tested alone. 

1.a Checkout a new release/<something> branch from this one.
1.b Comment out all the steps in `code_freeze` other than line 53 and 54. Commit the changes.
1.c `bundle install`
1.c Run `bundle exec fastlane code_freeze` and verify that `WooCommerce/metadata/release_notes.txt` has been updated with the content of the main `RELEASE_NOTES.txt` related to the current version.
1.d Delete the test branch

_2. Beta build_
2.a Checkout a new release/<something> branch from this one.
2.b Edit `build.gradle` with a new beta version and versionCode. Commit the changes.
2.c Run `bundle install`.
2.d Run `bundle exec fastlane build_and_upload_beta` and verify that the a release is drafted on GitHub and that the release notes are added, the "Pre-release" option is checked and the .apk is uploaded.
2.e Delete the artifact from Google Play.
2.f Delete the test branch.

_3. Release build_
2.a Checkout a new release/<something> branch from this one.
2.b Edit `build.gradle` with a final version and versionCode. Commit the changes.
2.c Run `bundle install`.
2.d Run `bundle exec fastlane build_and_upload_release` and verify that the a release is drafted on GitHub and that the release notes are added and the .apk and .aab are uploaded.
2.e Delete the artifact from Google Play.
2.f Delete the test branch.

**Note 1**: When this is ready to be merged, a new version of `release-toolkit` should be released and the references in this PR updated. 

**Note 2**: I'm targeting the release branch because this stuff is used only on releases and I think it's better to test it sooner than later :-) If the review won't be completed before the `4.5` release, I'll retarget it. 

**Note 3**: Related `release-tookit` PR: https://github.com/wordpress-mobile/release-toolkit/pull/136

 
Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mokagio commented:
> I manually tested code freeze and beta lanes. I'm getting a 401 from GitHub, but that must be an issue with my token. ~~The fact that I'm getting that error, though, at least shows me that the API request was made. I feel that's enough for me to approve.~~

That was a bit lazy of me, sorry. I went back and fixed the token.

The beta release was uploaded as described

<img width="1375" alt="Screen Shot 2020-06-25 at 11 32 06 pm" src="https://user-images.githubusercontent.com/1218433/85729937-942b2e00-b73c-11ea-9f14-39ccd4197eb9.png">

mokagio commented:
The release update lane also worked as described

<img width="1368" alt="Screen Shot 2020-06-25 at 11 54 54 pm" src="https://user-images.githubusercontent.com/1218433/85733318-714e4900-b73f-11ea-92eb-47f33f686044.png">

The Play Store upload wasn't compromised

<img width="994" alt="Screen Shot 2020-06-25 at 11 45 30 pm" src="https://user-images.githubusercontent.com/1218433/85733406-81febf00-b73f-11ea-9a48-2fb33d6ed2e6.png">

loremattei commented on fastlane/Fastfile:
```diff
@@ -51,6 +51,9 @@ SUPPORTED_LOCALES = [
     android_bump_version_release()
     new_version = android_get_app_version()
+    extract_release_notes_for_version(version: new_version, 
+      release_notes_file_path:"#{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.txt", 
+      extracted_notes_file_path:"#{ENV["PROJECT_ROOT_FOLDER"]}WooCommerce/metadata/release_notes.txt")
```
Right! We should! Maybe we should make it an action so that it can be shared across repositories (this is an issue we have in multiple places).

loremattei commented:
Hey @mokagio I've addressed your comments and updated the `release-toolkit` version, so this should be ready for another look. 

I agree with your idea about having a function to evaluate the path! I'd love if we could find a way to figure out if we are into a ruby command or into a call to shell and evaluate the right path for the case, but I think this isn't very easy to do. 
I'll try to build something in the next version of the `release-toolkit`.

oguzkocer commented:
While reviewing #2600, I came across this PR. It looks like this is merged to `release/4.6` but it has `4.5` as its milestone. I'll update that - let me know if I am missing something.
